addons/stock/models/stock_move.py

```diff
@@ -1018,14 +1018,18 @@ def _merge_moves(self, merge_into=False):
         :return: Recordset of moves passed to this method. If some of the passed moves were merged
         into another existing one, return this one and not the (now unlinked) original.
         """
-        distinct_fields = self._prepare_merge_moves_distinct_fields()
 
         candidate_moves_set = set()
         if not merge_into:
             self._update_candidate_moves_list(candidate_moves_set)
         else:
             candidate_moves_set.add(merge_into | self)
 
+        relevant_moves = self
+        for candidate_moves in candidate_moves_set:
+            relevant_moves |= candidate_moves
+        distinct_fields = relevant_moves._prepare_merge_moves_distinct_fields()
```

Comment:
can be slow to use `|` in a loop. you could rather do 

```suggestion
        distinct_fields = self | self.env['stock.move'].concat(*candidate_move_set)._prepare_merge_moves_distinct_fields()
```

Reply:
@Whenrow applied ! Thanks a lot for the review :slightly_smiling_face: